app/parse.py

```diff
@@ -8,5 +15,40 @@ class Course:
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+COURSE_FIELDS = [field.name for field in fields(Course)]
+
+
+def parse_single_course(page_soup: BeautifulSoup) -> Course:
+    return Course(name=page_soup.select_one("h3").text,
+                  short_description=page_soup.select_one(
+                      ".typography_landingTextMain__Rc8BD.mb-32"
+                  ).text,
+                  duration=page_soup.select_one(
+                      "p.typography_landingTextMain__Rc8BD.ProfessionCardTags_regularTag__yTc6K:last-of-type > span:last-of-type"
+                  ).text
+                  )
+
+
+def get_all_courses() -> [Course]:
+    page = requests.get(BASE_URL).content
```

Comment:
Consider adding error handling for the HTTP request. Use a try-except block to catch exceptions like `requests.exceptions.RequestException` to handle network-related errors gracefully.